### ISSUE TITLE
No tplot variables for ICON data via CDAWeb API

### ISSUE THREAD

**Issue by JonathonMSmith** [spedas/pyspedas#880]
When trying to load the ICON data using the CDAWeb API the data is not loaded into pytplot and every variable throws the error:
```
Could not process time variable 'Epoch' for the netcdf variable
```

To reproduce, the following code should work:

```
import pyspedas

cdaweb_obj = pyspedas.CDAWeb()

time0 = '2021-01-15T14:05:52'
time1 = '2021-01-15T15:08:57'

urllist = cdaweb_obj.get_filenames(['ICON_L2-2_MIGHTI_VECTOR-WIND-GREEN (2019-12-06 to 2022-11-25)'],time0, time1)

cdaweb_obj.cda_download(urllist,"cdaweb/",prefix='ICON_L22_')
```

Produces the following error (truncated for space):

```
11-Jun-24 16:48:24: Downloading https://cdaweb.gsfc.nasa.gov/sp_phys/data/icon/l2-2_mighti_vector-wind-green/2021/icon_l2-2_mighti_vector-wind-green_20210115_v05r001.nc to /home/jovyan/scratch_space/tutorial-notebooks/summer-school-24-main/pysat-tutorial/cdaweb///icon/l2-2_mighti_vector-wind-green/2021/icon_l2-2_mighti_vector-wind-green_20210115_v05r001.nc
11-Jun-24 16:48:24: Download complete: /home/jovyan/scratch_space/tutorial-notebooks/summer-school-24-main/pysat-tutorial/cdaweb///icon/l2-2_mighti_vector-wind-green/2021/icon_l2-2_mighti_vector-wind-green_20210115_v05r001.nc
11-Jun-24 16:48:24: Could not process time variable 'Epoch' for the netcdf variable: 'Epoch_Full'
11-Jun-24 16:48:24: Could not process time variable 'Epoch' for the netcdf variable: 'ICON_L22_Fringe_Amplitude'
11-Jun-24 16:48:24: Could not process time variable 'Epoch' for the netcdf variable: 'ICON_L22_Fringe_Amplitude_A'
...
11-Jun-24 16:48:24: Could not process time variable 'Epoch' for the netcdf variable: 'ICON_L22_Zonal_Wind_Precision_Low_Signal_Effect'
11-Jun-24 16:48:24: Downloaded 1 files.
11-Jun-24 16:48:24: tplot variables:
```

System info:
Python 3.10.14
pyspedas version: 1.5.12
macOS 14.5
working in a conda environment

**Comment by ericthewizard** [comment]
I looked into this a bit when Jonathon asked about it this morning, and it looks like the `time` keyword isn’t being passed to the netcdf_to_tplot function in cdaweb.py (line 236):

https://github.com/spedas/pyspedas/blob/2589660d47390a708cff46af11a63d1d7466a45e/pyspedas/cdagui/cdaweb.py#L236

**Comment by jameswilburlewis** [comment]
I think the situation is a bit more complicated than that.   The `time` keyword shouldn't be necessary.  ICON seems to be using some slightly nonstandard conventions in its NetCDF variable attributes ('Units' vs. 'units', 'ms' versus 'milliseconds since 1970-01-01 00:00:00').  This results in an exception handler in netcdf_to_tplot getting called, but only printing a generic error message without the exception details.

This can probably be fixed with a few changes to netcdf_to_tplot.

**Comment by jameswilburlewis** [comment]
This should now be fixed, in pyspedas 1.5.15 and pytplot-mpl-temp 2.2.27 .
